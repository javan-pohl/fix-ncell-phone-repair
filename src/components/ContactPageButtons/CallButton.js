import * as React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import { contactButton } from './ContactPage.module.css'
import phoneIcon from '../../images/call.png'

function getNum() {
  const data = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          phone
        }
      }
    }
  `)
  const {
    site: {
      siteMetadata: { phone },
    },
  } = data
  return phone
}

const CallButton = () => {
  const num = getNum()
  return (
    <button data-testid="callUsButton" className={contactButton} type="button">
      <a data-testid="callButtonA" href={`tel:${num}`}>
        <img src={phoneIcon} alt="email icon" />
        <br />
        Call or Text
      </a>
    </button>
  )
}

export default CallButton
