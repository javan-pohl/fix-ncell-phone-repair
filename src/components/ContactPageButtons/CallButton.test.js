import 'jsdom-global/register'
import '@testing-library/jest-dom'
import React from 'react'
import * as Gatsby from 'gatsby'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import CallButton from './CallButton'

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')
useStaticQuery.mockImplementation(() => ({
  site: {
    siteMetadata: {
      phone: '3034216499',
    },
  },
}))

beforeEach(() => {
  jest.clearAllMocks()
})

describe('CallButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CallButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should have href attribute "tel:3034216499"', () => {
    const { queryByTestId } = render(<CallButton />)
    const callButton = queryByTestId('callButtonA')
    expect(callButton).toHaveAttribute('href', 'tel:3034216499')
  })
})
