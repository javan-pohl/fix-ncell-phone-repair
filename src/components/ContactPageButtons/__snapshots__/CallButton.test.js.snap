// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallButton renders correctly 1`] = `
<button
  className="contactButton"
  data-testid="callUsButton"
  type="button"
>
  <a
    data-testid="callButtonA"
    href="tel:3034216499"
  >
    <img
      alt="email icon"
      src=""
    />
    <br />
    Call Us
  </a>
</button>
`;
